modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/RelatedObjectEntryResourceTest.java

```diff
@@ -0,0 +1,272 @@
+ */
+@RunWith(Arquillian.class)
+public class RelatedObjectEntryResourceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@BeforeClass
+	public static void setUpClass() {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-153324", "true"
+			).build());
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-153324", "false"
+			).build());
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_user = TestPropsValues.getUser();
+
+		_objectDefinition = _publishObjectDefinition(
+			Collections.singletonList(
+				ObjectFieldUtil.createObjectField(
+					"Text", "String", true, true, null,
+					RandomTestUtil.randomString(), _OBJECT_FIELD_NAME, false)));
+
+		_objectEntry = _createObjectEntry(_OBJECT_FIELD_VALUE);
+
+		_userSystemObjectDefinitionMetadata =
+			_systemObjectDefinitionMetadataTracker.
+				getSystemObjectDefinitionMetadata("User");
+
+		ObjectDefinition userSystemObjectDefinition =
+			_getSystemObjectDefinition(_userSystemObjectDefinitionMetadata);
+
+		_objectRelationship =
+			ObjectRelationshipLocalServiceUtil.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId(),
+				userSystemObjectDefinition.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		ObjectRelationshipLocalServiceUtil.
+			addObjectRelationshipMappingTableValues(
+				TestPropsValues.getUserId(),
+				_objectRelationship.getObjectRelationshipId(),
+				_objectEntry.getPrimaryKey(), _user.getUserId(),
+				ServiceContextTestUtil.getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ObjectRelationshipLocalServiceUtil.
+			deleteObjectRelationshipMappingTableValues(
+				_objectRelationship.getObjectRelationshipId(),
+				_objectEntry.getPrimaryKey(), _user.getUserId());
+		ObjectRelationshipLocalServiceUtil.deleteObjectRelationship(
+			_objectRelationship);
+	}
+
+	@Test
+	public void testGetSystemObjectRelatedObjects() throws Exception {
+		JSONObject jsonObject = _invoke(
+			StringBundler.concat(
+				_userSystemObjectDefinitionMetadata.getRESTContextPath(),
+				StringPool.SLASH, _user.getUserId(), StringPool.SLASH,
+				_objectRelationship.getName()),
+			Http.Method.GET);
+
+		JSONArray itemsJSONArray = jsonObject.getJSONArray("items");
+
+		Assert.assertEquals(1, itemsJSONArray.length());
+
+		JSONObject itemJSONObject = itemsJSONArray.getJSONObject(0);
+
+		Assert.assertEquals(
+			_OBJECT_FIELD_VALUE, itemJSONObject.getString(_OBJECT_FIELD_NAME));
+	}
+
+	@Test
+	public void testPutSystemObjectRelatedObjects() throws Exception {
```

Comment:
This test is only adding 1 related object. If so it should be named `testPutSystemObjectRelatedObjects`.

Also it would be great if you could assert that the relation is actually created in the database, not only that the response contains the field.

Reply:
Done!
I did it in a simple way making a request to the GET endpoint and checking if now there are 2 items. If you prefer something more explicit, I can do it.